sql/src/main/java/org/apache/druid/sql/calcite/expression/builtin/ArrayOffsetOperatorConversion.java

```diff
@@ -39,16 +38,15 @@
       .operandTypeChecker(
           OperandTypes.sequence(
               "(array,expr)",
-            OperandTypes.or(
-                OperandTypes.family(SqlTypeFamily.STRING),
-                OperandTypes.family(SqlTypeFamily.ARRAY),
-                OperandTypes.family(SqlTypeFamily.MULTISET)
-            ),
+              OperandTypes.or(
+                  OperandTypes.family(SqlTypeFamily.ARRAY),
+                  OperandTypes.family(SqlTypeFamily.STRING)
+              ),
             OperandTypes.family(SqlTypeFamily.NUMERIC)
```

Comment:
should be stricter - like having `SqlTypeFamily.INTEGER`?

Reply:
ah yeah, that probably could stand to be a bit further restricted